File: algorithm/merge-two-sorted-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* l1, ListNode* l2) {
        ListNode *head = new ListNode();
        ListNode *curr = head;
        
        while (l1 && l2){
            if (l2->val > l1->val){
                curr->next = l1;
                l1 = l1->next;
            }
            else {
                curr->next = l2;
                l2 = l2->next;
            }
            curr= curr->next;
                
        }
        
        curr->next = (l1 ? l1 : l2);
        curr = head->next;
        delete head;
        
        return curr;
    }
};

File: python/array and string/merge-two-sorted-lists.py
#https://leetcode.com/problems/merge-two-sorted-lists/

# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        if l1 == None and l2 == None:
            return None
        elif l1 == None:
            return l2
        elif l2 == None:
            return l1
        else:
            head = None
            if l1.val < l2.val:
                head = l1
                head.next = self.mergeTwoLists(l1.next, l2)
            else:
                head = l2
                head.next = self.mergeTwoLists(l1, l2.next)
            return head
                
        